lib/compiler_rt/atomics.zig

```diff
@@ -214,6 +225,31 @@ inline fn atomic_exchange_N(comptime T: type, ptr: *T, val: T, model: i32) T {
         const value = ptr.*;
         ptr.* = val;
         return value;
+    } else if (@sizeOf(T) < smallest_atomic_fetch_exch_size) {
+        // Machine does not support this type, but it does support a larger type.
+        const WideAtomic = std.meta.Int(.unsigned, smallest_atomic_fetch_exch_size * 8);
+
+        const addr = @ptrToInt(ptr);
+        const wide_addr = addr & ~(@as(T, smallest_atomic_fetch_exch_size) - 1);
```

Comment:
Could this accidentally trigger faults if the `wide_addr` drops into a page without read/write perms?

Reply:
Hm good question. I think as long as the wide type is less than the page size, there should be no problems. If the to-be-updated type is all the way at the end of a page, then `wide_addr` will point to exactly the last `WideAtomic` bytes before the page end. Similar on the other extreme.